kubernetes/resource_kubernetes_persistent_volume_v1_test.go

```diff
@@ -0,0 +1,2000 @@
+					resource.TestCheckResourceAttr(resourceName, "spec.0.persistent_volume_source.0.csi.0.node_publish_secret_ref.0.namespace", "default"),
+					resource.TestCheckResourceAttr(resourceName, "spec.0.persistent_volume_source.0.csi.0.controller_expand_secret_ref.0.name", name),
+					resource.TestCheckResourceAttr(resourceName, "spec.0.persistent_volume_source.0.csi.0.controller_expand_secret_ref.0.namespace", "default"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccKubernetesPersistentVolumeV1_volumeMode(t *testing.T) {
+	var conf api.PersistentVolume
+	name := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "kubernetes_persistent_volume_v1.test"
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		IDRefreshName:     resourceName,
+		IDRefreshIgnore:   []string{"metadata.0.resource_version"},
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckKubernetesPersistentVolumeV1Destroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccKubernetesPersistentVolumeV1VolumeModeConfig(name, "Block"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKubernetesPersistentVolumeV1Exists(resourceName, &conf),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.annotations.%", "2"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.annotations.TestAnnotationOne", "one"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.annotations.TestAnnotationTwo", "two"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.labels.%", "3"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.labels.TestLabelOne", "one"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.labels.TestLabelTwo", "two"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.labels.TestLabelThree", "three"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.name", name),
+					resource.TestCheckResourceAttrSet(resourceName, "metadata.0.generation"),
+					resource.TestCheckResourceAttrSet(resourceName, "metadata.0.resource_version"),
+					resource.TestCheckResourceAttrSet(resourceName, "metadata.0.uid"),
+					resource.TestCheckResourceAttr(resourceName, "spec.0.capacity.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "spec.0.capacity.storage", "5Gi"),
+					resource.TestCheckResourceAttr(resourceName, "spec.0.access_modes.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "spec.0.access_modes.0", "ReadWriteOnce"),
+					resource.TestCheckResourceAttr(resourceName, "spec.0.volume_mode", "Block"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccKubernetesPersistentVolumeV1_hostpath_claimRef(t *testing.T) {
+	var conf1, conf2 api.PersistentVolume
+	var conf3 api.PersistentVolumeClaim
+	randString := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+	name := fmt.Sprintf("tf-acc-test-%s", randString)
+
+	const resourceName = "kubernetes_persistent_volume_v1.test"
```

Comment:
Let's make this `resourceName := "kubernetes_persistent_volume_v1.test"`. For consistency.